pkg/agent/proxy/proxier.go

```diff
@@ -356,17 +377,15 @@ func (p *proxier) syncProxyRules() {
 		}
 		klog.V(4).Infof("syncProxyRules took %v", time.Since(start))
 	}()
-	if !p.isInitialized() {
-		klog.V(4).Info("Not syncing rules until both Services and Endpoints have been synced")
-		return
-	}
 
+	p.serviceEndpointsMapsMutex.Lock()
```

Comment:
I sugges to using a function to wrap this snippet.
```go
func() {
   lock()
   defer unlock()
   ...
}()
```

Reply:
Make sense. In this case, as really not much after the lock protection in this func, I just change to put "defer p.serviceEndpointsMapsMutex.Unlock()" after "p.serviceEndpointsMapsMutex.Lock()". Let me know what you think.

Reply:
Sounds good.